pkg/controller/replicaset/replica_set.go

```diff
@@ -591,31 +579,22 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 		return nil
 	}
 
+	// list all pods to include the pods that don't match the rs`s selector
```

Comment:
@enisoc do you mind leaving out the changes that are already in my PR that kargakis pointed above ?